components/tidb_query/src/batch/executors/top_n_executor.rs

```diff
@@ -775,4 +812,228 @@ mod tests {
         );
         assert!(r.is_drained.unwrap());
     }
+
+    /// Builds an executor that will return these data:
+    ///
+    /// == Schema ==
+    /// Col0 (Bytes[utf8_general_ci])      Col1(Bytes[utf8_bin])   Col2(Bytes[binary])
+    /// == Call #1 ==
+    /// "aa"                               "aaa"                      "カ"
+    /// NULL                               NULL                       "Aa"
+    /// "aa"                               "aa"                       NULL
+    /// == Call #2 ==
+    /// == Call #3 ==
+    /// "カ"                               "か"                        NULL
+    /// "か"                               "カ"                       "aa"
+    /// "Aa"                               NULL                       "aaa"
+    /// "aaa"                              "Aa"                       "か"
+    /// (drained)
+    fn make_bytes_src_executor() -> MockExecutor {
+        let mut col1: FieldType = FieldTypeTp::VarChar.into();
+        let mut col2: FieldType = FieldTypeTp::VarChar.into();
+        let mut col3: FieldType = FieldTypeTp::VarChar.into();
+        col1.set_collation(Collation::Utf8GeneralCi);
+        col2.set_collation(Collation::Utf8Bin);
+        col3.set_collation(Collation::Binary);
+        MockExecutor::new(
+            vec![col1, col2, col3],
+            vec![
+                BatchExecuteResult {
+                    physical_columns: LazyBatchColumnVec::from(vec![
+                        VectorValue::Bytes(vec![Some(b"aa".to_vec()), None, Some(b"aa".to_vec())]),
+                        VectorValue::Bytes(vec![Some(b"aa".to_vec()), None, Some(b"aaa".to_vec())]),
+                        VectorValue::Bytes(vec![
+                            None,
+                            Some(b"Aa".to_vec()),
+                            Some("カ".as_bytes().to_vec()),
```

Comment:
Kanas seem not controlled by GeneralCi. Is there any special reason to test with them?

Reply:
fixed